src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -238,8 +238,9 @@ def fetch_codecov_data(
         )
         # Get commit sha from Git blame if valid
         gh_integrations = integrations.filter(provider="github")
-        should_get_commit_sha = fetch_commit_sha and gh_integrations and not has_error_commit
```

Comment:
This is the bug.

The `sentry` project sometimes gets a commit from getsentry and sometimes from sentry.
If we get a commit from getsentry, we would not try to run `git blame` but try to get coverage data from the sentry but with getsentry's sha.